pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -1881,6 +1882,9 @@ class Timestamp(_Timestamp):
                                 hour or 0, minute or 0, second or 0, fold=fold or 0)
             unit = None
 
+        elif checknull_with_nat_and_na(ts_input):
+            return NaT
```

Comment:
can the 
```
        if ts.value == NPY_NAT:
            return NaT
```
be removed from below now?

Reply:
It doesn't look like it can be removed. Just tried it and this starts to raise if its removed:

`pd.Timestamp(np.datetime64("NaT"))`

Reply:
ah right, thanks, checknull checks for pandas NaT, not numpy NaT

does it work to do the check a bit deeper? i.e.
```diff
diff --git a/pandas/_libs/tslibs/conversion.pyx b/pandas/_libs/tslibs/conversion.pyx
index 57b7754b08..4d494d2282 100644
--- a/pandas/_libs/tslibs/conversion.pyx
+++ b/pandas/_libs/tslibs/conversion.pyx
@@ -1,4 +1,5 @@
 import numpy as np
+from pandas._libs.missing cimport checknull_with_nat_and_na
 
 cimport numpy as cnp
 from libc.math cimport log10
@@ -268,7 +269,7 @@ cdef _TSObject convert_to_tsobject(object ts, tzinfo tz, str unit,
     if isinstance(ts, str):
         return convert_str_to_tsobject(ts, tz, unit, dayfirst, yearfirst)
 
-    if ts is None or ts is NaT:
+    if checknull_with_nat_and_na(ts):
         obj.value = NPY_NAT
     elif is_datetime64_object(ts):
         reso = get_supported_reso(get_datetime64_unit(ts))
diff --git a/pandas/_libs/tslibs/timestamps.pyx b/pandas/_libs/tslibs/timestamps.pyx
index 80b7c2d2f5..59640a3ef4 100644
--- a/pandas/_libs/tslibs/timestamps.pyx
+++ b/pandas/_libs/tslibs/timestamps.pyx
@@ -1882,9 +1882,6 @@ class Timestamp(_Timestamp):
                                 hour or 0, minute or 0, second or 0, fold=fold or 0)
             unit = None
 
-        elif checknull_with_nat_and_na(ts_input):
-            return NaT
-
         if getattr(ts_input, "tzinfo", None) is not None and tz is not None:
             raise ValueError("Cannot pass a datetime or Timestamp with tzinfo with "
                              "the tz parameter. Use tz_convert instead.")
```
?

Reply:
yes - updated. thanks